scripts/setup/generate_secrets.py

```diff
@@ -180,6 +184,20 @@ def add_secret(name: str, value: str) -> None:
     if need_secret("zulip_org_id"):
         add_secret("zulip_org_id", str(uuid.uuid4()))
 
+    if need_secret("push_registration_encryption_keys"):
+        # We defer importing settings unless we need it, because
+        # importing settings is expensive (mostly because of
+        # django-auth-ldap) and we want the noop case to be fast.
+        from zproject import settings
+
+        if settings.ZILENCER_ENABLED:
+            private_key = PrivateKey.generate()
+            private_key_str = Base64Encoder.encode(bytes(private_key)).decode("utf-8")
+            public_key_str = Base64Encoder.encode(bytes(private_key.public_key)).decode("utf-8")
+            add_secret(
+                "push_registration_encryption_keys", json.dumps({public_key_str: private_key_str})
```

Comment:
This JSON pattern inside a single secret is a bit unusual; is there a reason to prefer it to a pair of settings like we have with (`zulip_org_id`/`zulip_org_key`)? 

I think the reason folks were talking about a JSON file is they were imagining a separate file off the side; but it probably does make sense to put this in the normal secrets file, in which case I think just two secrets is reasonable.

Reply:
> This JSON pattern inside a single secret is a bit unusual; is there a reason to prefer it to a pair of settings like we have with (zulip_org_id/zulip_org_key)?

We plan to support multiple public-private key pair at once. 

The ZAP says:
> The push bouncer maintains a push_registration_encryption_keys map containing asymmetric key pairs ({public_key: private_key}), each generated to support key rotation. When a new key pair is created, the public key (bouncer_public_key) is baked into a new client release.

> To support older clients, the bouncer continues to accept the older public keys for a period of time, allowing clients to upgrade. Once a sufficient portion of clients have updated, the corresponding old keys are removed from the map.

Reply:
Yeah that seems fine for now. Easy to migrate later, since this is zilencer side configuration only.